{"info": {"author": "Plone Security Team", "author_email": "security@plone.org", "bugtrack_url": null, "classifiers": ["Development Status :: 5 - Production/Stable", "Framework :: Plone", "Framework :: Plone :: 4.0", "Framework :: Plone :: 4.1", "Framework :: Plone :: 4.2", "Framework :: Plone :: 4.3", "Framework :: Plone :: 5.0", "Framework :: Plone :: 5.1", "Framework :: Plone :: 5.2", "License :: OSI Approved :: GNU General Public License (GPL)", "Programming Language :: Python", "Programming Language :: Python :: 2.6", "Programming Language :: Python :: 2.7", "Programming Language :: Python :: 3.6", "Programming Language :: Python :: 3.7"], "description": "Plone hotfix, 2020-01-21\n========================\n\nThis hotfix fixes several security issues:\n\n- Privilege escalation when ``plone.restapi`` is installed.\n  Reported and fixed by Lukas Graf and Niklaus Johner.\n\n- An open redirection on the login form and possibly other places where redirects are done.\n  The ``isURLInPortal`` check that is done to avoid linking to an external site could be tricked into accepting malicious links.\n  Reported by Damiano Esposito.\n\n- Password strength checks were not always checked.\n  Reported by Ben Kummer.\n\n- You might be able to PUT (overwrite) some content without needing write permission.\n  This seems hard to do in practice.\n  This fix is only needed when you use ``plone.app.contenttypes``.\n  Reported and fixed by Alessandro Pisa.\n\n- SQL quoting in DTML or in connection objects was insufficient, leading to possible SQL injections.\n  This is a problem in Zope.  If you use Zope without Plone, this hotfix should work for you too.\n  Reported and fixed by Michael Brunnbauer and Michael Howitz.\n\n- Cross Site Scripting (XSS) in the title field on plone 5.0 and higher.\n  Reported by Marcos Valle.\n\n\nCompatibility\n=============\n\nThis hotfix should be applied to the following versions of Plone:\n\n* Plone 5.2.1 and any earlier 5.2.x version\n* Plone 5.1.6 and any earlier 5.x version\n* Plone 4.3.19 and any earlier 4.x version\n\nThe hotfix is officially supported by the Plone security team on the\nfollowing versions of Plone in accordance with the Plone\n`version support policy <https://plone.org/security/update-policy>`_: 4.3.19, 5.0.10, 5.1.6 and 5.2.1.\n\nOn Plone 4.3, 5.0 and 5.1 the hotfix is officially only supported on Python 2.7.\nOn Plone 5.2.X it is supported on Python 2.7 and Python 3.6/3.7.\n\nPython 2.6 should work, and earlier Plone 4 versions should work too, but this has not been fully tested.\n\nThe fixes included here will be incorporated into subsequent releases of Plone,\nso Plone 4.3.20, 5.1.7, 5.2.2 and greater should not require this hotfix.\n\n\nInstallation\n============\n\nInstallation instructions can be found at\nhttps://plone.org/security/hotfix/20200121\n\n.. note::\n\n  You may get an error when running buildout::\n\n    Error: Couldn't find a distribution for 'Products.PloneHotfix20200121==1.0'.\n\n  The most likely cause is that your buildout is trying to download the hotfix via http or from an older PyPI index.\n  In the buildout section of your buildout, make sure you use the correct index::\n\n    [buildout]\n    index = https://pypi.org/simple/\n\n\nQ&A\n===\n\nQ: How can I confirm that the hotfix is installed correctly and my site is protected?\n  A: On startup, the hotfix will log a number of messages to the Zope event log\n  that look like this::\n\n      2020-01-21 13:10:26 INFO Products.PloneHotfix20200121 Applied sql_quote patch\n      2020-01-21 13:10:26 INFO Products.PloneHotfix20200121 Applied in_portal patch\n      2020-01-21 13:10:26 INFO Products.PloneHotfix20200121 Applied password_validation patch\n      2020-01-21 13:10:26 INFO Products.PloneHotfix20200121 Applied pac patch\n      2020-01-21 13:10:26 INFO Products.PloneHotfix20200121 Applied content patch\n      2020-01-21 13:10:26 INFO Products.PloneHotfix20200121 Applied layout patch\n      2020-01-21 13:10:26 INFO Products.PloneHotfix20200121 Applied restapi_local_roles patch\n      2020-01-21 13:10:26 INFO Products.PloneHotfix20200121 Hotfix installed\n\n  The exact number of patches applied, will differ depending on what packages you are using.\n  If a patch is attempted but fails, it will be logged as an error that says\n  \"Could not apply\". This may indicate that you have a non-standard Plone\n  installation.  Please investigate, and mail us the accompanying traceback if you think it is a problem in the hotfix.\n\nQ: How can I report problems installing the patch?\n  A: Contact the Plone security team at security@plone.org or visit the\n  Gitter channel at https://gitter.im/plone/public and the forum at https://community.plone.org\n\nQ: How can I report other potential security vulnerabilities?\n  A: Please email the security team at security@plone.org rather than discussing\n  potential security issues publicly.\n\nChangelog\n=========\n\n\n1.1 (2020-02-11)\n----------------\n\n- Improve the ``sql_quote`` fix.\n  For details and discussion, see `DocumentTemplate <https://github.com/zopefoundation/DocumentTemplate/issues/48>`_.\n  If you are not using SQL in your website you do **NOT** need to upgrade.\n  (You can if you want to.)\n  Default Plone does not need it.\n  Upgrading to this version is especially recommended when you use Postgres.\n  Note that ``RelStorage`` is not affected.\n  Specific changes:\n\n  - In ``sql_quote`` no longer escape double quotes and backslashes by doubling them:\n    on MySQL this was fine, but on Postgres valid double quotes and backslashes\n    would actually end up twice in the database.\n\n  - In ``sql_quote`` remove the special ``\\x1a`` NUL character and the carriage return ``\\r``.\n    This change is good for both MySQL and Postgres.\n\n\n1.0 (2020-01-21)\n----------------\n\n- Initial release\n\n\n", "description_content_type": "", "docs_url": null, "download_url": "", "downloads": {"last_day": -1, "last_month": -1, "last_week": -1}, "home_page": "https://plone.org/security/hotfix/20200121", "keywords": "plone security hotfix patch", "license": "GPL", "maintainer": "", "maintainer_email": "", "name": "Products.PloneHotfix20200121", "package_url": "https://pypi.org/project/Products.PloneHotfix20200121/", "platform": "", "project_url": "https://pypi.org/project/Products.PloneHotfix20200121/", "project_urls": {"Homepage": "https://plone.org/security/hotfix/20200121"}, "release_url": "https://pypi.org/project/Products.PloneHotfix20200121/1.1/", "requires_dist": ["setuptools"], "requires_python": "", "summary": "Various Plone hotfixes, 2020-01-21", "version": "1.1", "yanked": false, "html_description": "<div class=\"project-description\">\n            <div id=\"plone-hotfix-2020-01-21\">\n<h2>Plone hotfix, 2020-01-21</h2>\n<p>This hotfix fixes several security issues:</p>\n<ul>\n<li>Privilege escalation when <tt>plone.restapi</tt> is installed.\nReported and fixed by Lukas Graf and Niklaus Johner.</li>\n<li>An open redirection on the login form and possibly other places where redirects are done.\nThe <tt>isURLInPortal</tt> check that is done to avoid linking to an external site could be tricked into accepting malicious links.\nReported by Damiano Esposito.</li>\n<li>Password strength checks were not always checked.\nReported by Ben Kummer.</li>\n<li>You might be able to PUT (overwrite) some content without needing write permission.\nThis seems hard to do in practice.\nThis fix is only needed when you use <tt>plone.app.contenttypes</tt>.\nReported and fixed by Alessandro Pisa.</li>\n<li>SQL quoting in DTML or in connection objects was insufficient, leading to possible SQL injections.\nThis is a problem in Zope.  If you use Zope without Plone, this hotfix should work for you too.\nReported and fixed by Michael Brunnbauer and Michael Howitz.</li>\n<li>Cross Site Scripting (XSS) in the title field on plone 5.0 and higher.\nReported by Marcos Valle.</li>\n</ul>\n</div>\n<div id=\"compatibility\">\n<h2>Compatibility</h2>\n<p>This hotfix should be applied to the following versions of Plone:</p>\n<ul>\n<li>Plone 5.2.1 and any earlier 5.2.x version</li>\n<li>Plone 5.1.6 and any earlier 5.x version</li>\n<li>Plone 4.3.19 and any earlier 4.x version</li>\n</ul>\n<p>The hotfix is officially supported by the Plone security team on the\nfollowing versions of Plone in accordance with the Plone\n<a href=\"https://plone.org/security/update-policy\" rel=\"nofollow\">version support policy</a>: 4.3.19, 5.0.10, 5.1.6 and 5.2.1.</p>\n<p>On Plone 4.3, 5.0 and 5.1 the hotfix is officially only supported on Python 2.7.\nOn Plone 5.2.X it is supported on Python 2.7 and Python 3.6/3.7.</p>\n<p>Python 2.6 should work, and earlier Plone 4 versions should work too, but this has not been fully tested.</p>\n<p>The fixes included here will be incorporated into subsequent releases of Plone,\nso Plone 4.3.20, 5.1.7, 5.2.2 and greater should not require this hotfix.</p>\n</div>\n<div id=\"installation\">\n<h2>Installation</h2>\n<p>Installation instructions can be found at\n<a href=\"https://plone.org/security/hotfix/20200121\" rel=\"nofollow\">https://plone.org/security/hotfix/20200121</a></p>\n<div>\n<p>Note</p>\n<p>You may get an error when running buildout:</p>\n<pre>Error: Couldn't find a distribution for 'Products.PloneHotfix20200121==1.0'.\n</pre>\n<p>The most likely cause is that your buildout is trying to download the hotfix via http or from an older PyPI index.\nIn the buildout section of your buildout, make sure you use the correct index:</p>\n<pre>[buildout]\nindex = https://pypi.org/simple/\n</pre>\n</div>\n</div>\n<div id=\"q-a\">\n<h2>Q&amp;A</h2>\n<dl>\n<dt>Q: How can I confirm that the hotfix is installed correctly and my site is protected?</dt>\n<dd><p>A: On startup, the hotfix will log a number of messages to the Zope event log\nthat look like this:</p>\n<pre>2020-01-21 13:10:26 INFO Products.PloneHotfix20200121 Applied sql_quote patch\n2020-01-21 13:10:26 INFO Products.PloneHotfix20200121 Applied in_portal patch\n2020-01-21 13:10:26 INFO Products.PloneHotfix20200121 Applied password_validation patch\n2020-01-21 13:10:26 INFO Products.PloneHotfix20200121 Applied pac patch\n2020-01-21 13:10:26 INFO Products.PloneHotfix20200121 Applied content patch\n2020-01-21 13:10:26 INFO Products.PloneHotfix20200121 Applied layout patch\n2020-01-21 13:10:26 INFO Products.PloneHotfix20200121 Applied restapi_local_roles patch\n2020-01-21 13:10:26 INFO Products.PloneHotfix20200121 Hotfix installed\n</pre>\n<p>The exact number of patches applied, will differ depending on what packages you are using.\nIf a patch is attempted but fails, it will be logged as an error that says\n\u201cCould not apply\u201d. This may indicate that you have a non-standard Plone\ninstallation.  Please investigate, and mail us the accompanying traceback if you think it is a problem in the hotfix.</p>\n</dd>\n<dt>Q: How can I report problems installing the patch?</dt>\n<dd>A: Contact the Plone security team at <a href=\"mailto:security%40plone.org\">security<span>@</span>plone<span>.</span>org</a> or visit the\nGitter channel at <a href=\"https://gitter.im/plone/public\" rel=\"nofollow\">https://gitter.im/plone/public</a> and the forum at <a href=\"https://community.plone.org\" rel=\"nofollow\">https://community.plone.org</a></dd>\n<dt>Q: How can I report other potential security vulnerabilities?</dt>\n<dd>A: Please email the security team at <a href=\"mailto:security%40plone.org\">security<span>@</span>plone<span>.</span>org</a> rather than discussing\npotential security issues publicly.</dd>\n</dl>\n</div>\n<div id=\"changelog\">\n<h2>Changelog</h2>\n<div id=\"id1\">\n<h3>1.1 (2020-02-11)</h3>\n<ul>\n<li>Improve the <tt>sql_quote</tt> fix.\nFor details and discussion, see <a href=\"https://github.com/zopefoundation/DocumentTemplate/issues/48\" rel=\"nofollow\">DocumentTemplate</a>.\nIf you are not using SQL in your website you do <strong>NOT</strong> need to upgrade.\n(You can if you want to.)\nDefault Plone does not need it.\nUpgrading to this version is especially recommended when you use Postgres.\nNote that <tt>RelStorage</tt> is not affected.\nSpecific changes:<ul>\n<li>In <tt>sql_quote</tt> no longer escape double quotes and backslashes by doubling them:\non MySQL this was fine, but on Postgres valid double quotes and backslashes\nwould actually end up twice in the database.</li>\n<li>In <tt>sql_quote</tt> remove the special <tt>\\x1a</tt> NUL character and the carriage return <tt>\\r</tt>.\nThis change is good for both MySQL and Postgres.</li>\n</ul>\n</li>\n</ul>\n</div>\n<div id=\"id2\">\n<h3>1.0 (2020-01-21)</h3>\n<ul>\n<li>Initial release</li>\n</ul>\n</div>\n</div>\n\n          </div>"}, "last_serial": 6613052, "releases": {"1.0": [{"comment_text": "", "digests": {"md5": "06cbb2ba08e844f73a40fb55e7cff1dc", "sha256": "a10f485984fb28a704cc13ef147dafb0f46c56eef2e56a03f9222b39edfb0b89"}, "downloads": -1, "filename": "Products.PloneHotfix20200121-1.0-py2.py3-none-any.whl", "has_sig": false, "md5_digest": "06cbb2ba08e844f73a40fb55e7cff1dc", "packagetype": "bdist_wheel", "python_version": "py2.py3", "requires_python": null, "size": 27674, "upload_time": "2020-01-21T15:00:05", "upload_time_iso_8601": "2020-01-21T15:00:05.259305Z", "url": "https://files.pythonhosted.org/packages/66/42/21664a56f5fe50ddc33add9eb681483a1d901a5d08832ec16b6ca33c608b/Products.PloneHotfix20200121-1.0-py2.py3-none-any.whl", "yanked": false}, {"comment_text": "", "digests": {"md5": "c9c643e4f71e50356433d4baa408d319", "sha256": "b82c103c36ddcfe1528aee4e501e1230d055e6d604155e8b612b075ad8c4edab"}, "downloads": -1, "filename": "Products.PloneHotfix20200121-1.0.tar.gz", "has_sig": false, "md5_digest": "c9c643e4f71e50356433d4baa408d319", "packagetype": "sdist", "python_version": "source", "requires_python": null, "size": 17203, "upload_time": "2020-01-21T15:00:06", "upload_time_iso_8601": "2020-01-21T15:00:06.733004Z", "url": "https://files.pythonhosted.org/packages/bc/c6/81f5428a13bd5cc92003d5a7a1057fb24e90d7358f8b38c205f7dbe5f74a/Products.PloneHotfix20200121-1.0.tar.gz", "yanked": false}], "1.1": [{"comment_text": "", "digests": {"md5": "289652d191367cdb098e2c773cb7340e", "sha256": "78fcd2132df95ba290d2469d552a7c4421ff5c6b66fad59e19e3bc1b5f179c6a"}, "downloads": -1, "filename": "Products.PloneHotfix20200121-1.1-py2.py3-none-any.whl", "has_sig": false, "md5_digest": "289652d191367cdb098e2c773cb7340e", "packagetype": "bdist_wheel", "python_version": "py2.py3", "requires_python": null, "size": 28529, "upload_time": "2020-02-11T22:47:25", "upload_time_iso_8601": "2020-02-11T22:47:25.440539Z", "url": "https://files.pythonhosted.org/packages/ee/2b/40106603d4d22fa3189bdcfe396c93ddccfb225be5e851f4db4bc49ebc7b/Products.PloneHotfix20200121-1.1-py2.py3-none-any.whl", "yanked": false}, {"comment_text": "", "digests": {"md5": "f819eaaf60643a794be73d96d9c9bb72", "sha256": "9f4c55f75aa9a2e15d778a514cf38d7134fe248eabc364f75680941194f5b1df"}, "downloads": -1, "filename": "Products.PloneHotfix20200121-1.1.tar.gz", "has_sig": false, "md5_digest": "f819eaaf60643a794be73d96d9c9bb72", "packagetype": "sdist", "python_version": "source", "requires_python": null, "size": 17950, "upload_time": "2020-02-11T22:47:27", "upload_time_iso_8601": "2020-02-11T22:47:27.157541Z", "url": "https://files.pythonhosted.org/packages/b3/f7/4054119d5a9cd92d274e5d86ee81761792ae7a3e0a5e1a0a01e78e139c3e/Products.PloneHotfix20200121-1.1.tar.gz", "yanked": false}]}, "urls": [{"comment_text": "", "digests": {"md5": "289652d191367cdb098e2c773cb7340e", "sha256": "78fcd2132df95ba290d2469d552a7c4421ff5c6b66fad59e19e3bc1b5f179c6a"}, "downloads": -1, "filename": "Products.PloneHotfix20200121-1.1-py2.py3-none-any.whl", "has_sig": false, "md5_digest": "289652d191367cdb098e2c773cb7340e", "packagetype": "bdist_wheel", "python_version": "py2.py3", "requires_python": null, "size": 28529, "upload_time": "2020-02-11T22:47:25", "upload_time_iso_8601": "2020-02-11T22:47:25.440539Z", "url": "https://files.pythonhosted.org/packages/ee/2b/40106603d4d22fa3189bdcfe396c93ddccfb225be5e851f4db4bc49ebc7b/Products.PloneHotfix20200121-1.1-py2.py3-none-any.whl", "yanked": false}, {"comment_text": "", "digests": {"md5": "f819eaaf60643a794be73d96d9c9bb72", "sha256": "9f4c55f75aa9a2e15d778a514cf38d7134fe248eabc364f75680941194f5b1df"}, "downloads": -1, "filename": "Products.PloneHotfix20200121-1.1.tar.gz", "has_sig": false, "md5_digest": "f819eaaf60643a794be73d96d9c9bb72", "packagetype": "sdist", "python_version": "source", "requires_python": null, "size": 17950, "upload_time": "2020-02-11T22:47:27", "upload_time_iso_8601": "2020-02-11T22:47:27.157541Z", "url": "https://files.pythonhosted.org/packages/b3/f7/4054119d5a9cd92d274e5d86ee81761792ae7a3e0a5e1a0a01e78e139c3e/Products.PloneHotfix20200121-1.1.tar.gz", "yanked": false}], "timestamp": "Fri May  8 03:17:58 2020"}